docs/prover/techniques/index.md

```diff
@@ -1,31 +1,121 @@
+            else
+                [split_1, split_2] = split_single(cfg)
+                worklist.add([split_1, current_depth + 1])
+                worklist.add([split_2, current_depth + 1])
+return UNSAT
+```
+
+Intuitively, the algorithm explores the tree of all possible recursive
+splittings along a fixed sequence of split points up to the maximum splitting
+depth. We call the splits at maximum splitting depth *split leafs*. The
+exploration stops in either of the following three cases:
+ - if one split was found that is SAT (reasoning: if one split is SAT, then the
+  original program must be SAT, since the behavior of the split is replayable in
+  the original program)
+ - if all splits have been shown to be unsat
+ - if solving on a split leaf has timed out (except if 
+   {ref}`-dontStopAtFirstSplitTimeout` has been set)
+
+The settings with which the user can influence this process are the
+following (each links to a more detailed description of the option):
 
-% TODO write this -- tracked in https://certora.atlassian.net/browse/DOC-351
+ - [Maximum split depth](-depth) controls the maximum splitting depth.
+ - [Medium timeout](-mediumTimeout) controls the timeout that is applied when
+   checking splits that are not split leafs, i.e., that are not at the maximum
+   depth. 
+ - [Smt timeout](--smt_timeout) controls the timeout that is used to solve split 
+   leafs; if this is exceeded, the Prover will give up with a TIMEOUT 
+   result, unless [the corresponding setting](-dontStopAtFirstSplitTimeout) says 
+   to go on.
+ - Setting the [initial splitting depth](-smt_initialSplitDepth) to a value 
+   above 0 will make the Prover skip the checking and immediately enumerate all 
+   splits up to that depth.
 
 (storage-and-memory-analysis)=
-# Storage and Memory Analysis
+# Storage and Memory analysis
 
 The Certora Prover works on EVM bytecode as its input. To the bytecode, the
 address space of both Storage and Memory are flat number lines. That two
 contract fields `x` and `y` don't share the same memory is an arithmetic
 property. With more complex data structures like mappings, arrays, and structs,
 this means that every
 ["non-aliasing"](https://en.wikipedia.org/wiki/Aliasing_(computing)) argument
-requires reasoning about multiplications, additions, and hash functions. Certora
-Prover models this reasoning correctly, but this naive low-level modeling can
-quickly overwhelm SMT solvers. In order to handle storage efficiently, Certora
-Prover analyses Storage (Memory) accesses in EVM code in order to understand the
-Storage (Memory) layout, thus making information like "an update to mapping `x`
-will never overwrite the scalar variable `y`" much more obvious to the SMT
-solvers. For scaling SMT solving to larger programs, these simplifications are
-essential.
+requires reasoning about multiplications, additions, and hash functions. The
+Certora Prover models this reasoning correctly, but this naive low-level
```

Comment:
I have some additional edits for this section but I see this PR doesn't really modify this section so I will address them another time.